fuzzy finder: add docs

taras-yemets opened the pull request:
Adds fuzzy finder docs.

![image](https://user-images.githubusercontent.com/25318659/206425900-6bc41017-cae0-401c-9fa9-7895dcead381.png)

Adding fuzzy finder to the code search explanation section instead of code-nav explanation section as in my opinion it better fits the rest of the content (cc: @felixfbecker). 

## Test plan
Tested manually.

<!-- All pull requests REQUIRE a test plan: https://docs.sourcegraph.com/dev/background-information/testing_principles -->

felixfbecker commented on doc/code_search/explanations/features.md:
```diff
@@ -98,6 +98,13 @@ Your site admin can enable search contexts on your private instance in **global
+Fuzzy finder helps you find any item by its name in the current repository or everywhere. You can search for repositories, symbols, files and files. 
+To open the fuzzy finder, press `Cmd/Ctrl+K` anywhere on the Sourcegraph web app.
+
+<img src="https://i.imgur.com/ErZOOMG.png" alt="Fuzzy search">
```
Could we use our `sourcegraph-assets` GCS bucket instead of imgur?

felixfbecker reviewed (commented):
👏🏻 for adding docs.

olafurpg commented on doc/code_search/explanations/features.md:
```diff
@@ -98,6 +98,13 @@ Your site admin can enable search contexts on your private instance in **global
+## Fuzzy search
+
+Fuzzy finder helps you find any item by its name in the current repository or everywhere. You can search for repositories, symbols, files and files. 
+To open the fuzzy finder, press `Cmd/Ctrl+K` anywhere on the Sourcegraph web app.
```
```suggestion
To open the fuzzy finder, press `Cmd+K` (macOS) or `Ctrl+K` (Linux/Windows) from any page.
```

olafurpg replied:
Use the dedicated Repos, Symbols, and Files tabs to search only for a repository, symbol, or file. Each tab has a dedicated shortcut:

* Repos: Cmd+I (macOS), Ctrl+K (Linux/Windows)
* Symbols: Cmd+O (macOS), Cmd+Shift+O (macOS Safari), Ctrl+O (Linux/Windows)
* Files: Cmd+P (macOS), Ctrl+P (Linux/Windows)

olafurpg commented on doc/code_search/explanations/features.md:
```diff
@@ -98,6 +98,13 @@ Your site admin can enable search contexts on your private instance in **global
 
+## Fuzzy search
+
+Fuzzy finder helps you find any item by its name in the current repository or everywhere. You can search for repositories, symbols, files and files. 
```
```suggestion
Use the fuzzy finder to quickly navigate to a repository, symbol, or file.
```

olafurpg commented on doc/code_search/explanations/features.md:
```diff
@@ -98,6 +98,13 @@ Your site admin can enable search contexts on your private instance in **global
+To open the fuzzy finder, press `Cmd/Ctrl+K` anywhere on the Sourcegraph web app.
+
+<img src="https://i.imgur.com/ErZOOMG.png" alt="Fuzzy search">
+
```
Use the "Searching everywhere" or "Searching in this repo" filter to determine whether to search for results only in the active repository or globally.

SCREENSHOT.

olafurpg reviewed (changes requested):
+1 to @felixfbecker's comment along with few other changes

taras-yemets commented:
@felixfbecker, @olafurpg, 
Thank you for your feedback!

Addressed it and added experimental badge with a note linking to the [feedback issue](https://github.com/sourcegraph/sourcegraph/discussions/42874).

<img width="885" alt="Screenshot 2022-12-12 at 14 48 38" src="https://user-images.githubusercontent.com/25318659/207050116-bcfea207-17f3-40f9-8d6b-6b2e04ee46ee.png">
